docs/developer/design/volume_v1.0.rst

```diff
@@ -0,0 +1,298 @@
+* Replaces:
+* Superseded-By:
+
+
+Absract
+-------
+
+To be able to run stateful services (such as Prometheus, Zenko or Hyperdrive),
+MetalK8s needs the ability to provide and manage persistent storage resources.
+
+To do so we introduce the concept of MetalK8s **Volume**, using a **Custom
+Resource Definition** (CRD), built on top of the existing concept of
+**Persistent Volume** from Kubernetes. Those **Custom Resources** (CR) will be
+managed by a dedicated Kubernetes operator which will be responsible for the
+storage preparation (using Salt states) and lifetime management of the backing
+**Persistent Volume**.
+
+Volume management will be available from the Platform UI (through a dedicated
+tab under the Node page). There, users will be able to create, monitor and
+delete MetalK8s volumes.
+
+
+Scope
+-----
+
+The scope of this first version of Volume Management will be minimalist but
+still functionally useful.
+
+
+Goals
+^^^^^
+
+* support two kinds of **Volume**:
+
+  * **sparseLoopDevice** (backed by a sparse file)
+  * **rawBlockDevice** (using whole disk)
+
+* add support for volume creation (one by one) in the Platform UI
+* add support for volume deletion (one by one) in the Platform UI
+* add support for volume listing/monitoring (show status, size, …) in the
+  Platform UI
+* document how to create a volume
+* document how to create a **StorageClass** object
+* automated tests on volume workflow (creation, deletion, …)
+
+
+Non-Goals
+^^^^^^^^^
+
+* RAID support
+* LVM support
+* expose raw block device (unformated) as **Volume**
+* use an **Admission Controller** for semantic validation
+* auto-discovery of the disks
+* batch provisioning from the Platform UI
+
+
+Proposal
+--------
+
+To implement this feature we need to:
+
+* define and deploy a new CRD describing a MetalK8s **Volume**
+* develop and deploy a new Kubernetes operator to manage the MetalK8s volumes
+* develop new Salt states to prepare and cleanup underlying storage on the
+  nodes
+* update the Platform UI to allow volume management
+
+
+User Stories
+^^^^^^^^^^^^
+
+
+Volume Creation
+~~~~~~~~~~~~~~~
+
+As a user I need to be able to create MetalK8s volume from the Platform UI.
+
+At creation time I can specify the type of volume I want, and then either its
+size (for **sparseLoopDevice**) or the backing device (for **rawBlockDevice**).
+
+I should be able monitor the progress of the volume creation from the Platform
+UI and see when the volume is ready to use (or if an error occured).
+
+
+Volume Monitoring
+~~~~~~~~~~~~~~~~~
+
+As a user I should be able to see all the volumes existing on a specified node
+as well as their states.
+
+
+Volume Deletion
+~~~~~~~~~~~~~~~
+
+As a user I need to be able to delete a MetalK8s volume from the Platform UI
+when I no longer need it.
+
+The Platform UI should prevent me from deleting Volumes in use.
+
+I should be able monitor the progress of the volume deletion from the Platform
+UI.
+
+
+Components Interactions
```

Comment:
```suggestion
Component Interactions
```
not sure though

Reply:
I agree.

Reply:
Why `lifecycle` over `interactions`?

Reply:
Because I'm struggling with how to write this in a non-awkward fashion :p I'm not sure which is the correct way to pluralize (or not) this title.

With a rested brain @nootal 's way seems correct

Reply:
Done